web_notify/static/src/xml/notification.xml

```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates id="template" xml:space="preserve">
+
+<t t-name="Notification.reload">
+    <div class="o_notification_action" t-if="widget.options.show_reload">
+        <a href="#" class="o_notification_reload_view">
+            <span class="fa fa-refresh"/> Reload current view
+        </a>
+    </div>
+</t>
+
+<t t-name="Notification.do_action">
+    <div class="o_notification_action" t-if="widget.options.action">
+        <a href="#" class="o_notification_do_action">
+            <span t-att-class="'fa ' + (widget.options.action_fa_icon ? widget.options.action_fa_icon : 'fa-arrow-circle-left')"/>
```

Comment:
Are the `t-att-class` escaped? If not, a jsonrpc call could allow to inject something in widget.options.action_fa_icon that would directly open a notification to a user with a crafted payload, creepy.

Reply:
[`t-att`](https://github.com/odoo/odoo/blob/39f0f5db77f793c7fb996d53195524de0e035de4/odoo/addons/base/ir/ir_qweb/qweb.py#L886
) uses  [`_compile_expr`](https://github.com/odoo/odoo/blob/10.0/odoo/addons/base/ir/ir_qweb/ir_qweb.py#L293).
Seems to be safe on eval but not sure it is for escape.
Not sure how that could be dangerous on a class tho...

In any case that would come via this api so why don't you escape it in _notify_channel if present?